Common/src/drivers/CDriverBase.cpp

```diff
@@ -0,0 +1,1393 @@
+}
+
+void CDriverBase::SetMarkerVelocities(unsigned short iMarker, vector<vector<passivedouble>> values) {
+  if (!main_config->GetDeform_Mesh()) {
+    SU2_MPI::Error("Mesh solver is not defined!", CURRENT_FUNCTION);
+  }
+
+  const auto nVertex = GetNumberMarkerVertices(iMarker);
+
+  if (values.size() != nVertex) {
+    SU2_MPI::Error("Invalid number of marker vertices!", CURRENT_FUNCTION);
+  }
+
+  for (auto iVertex = 0ul; iVertex < nVertex; iVertex++) {
+    SetMarkerVelocities(iMarker, iVertex, values[iVertex]);
+  }
+}
+
+void CDriverBase::SetMarkerVelocities(unsigned short iMarker, unsigned long iVertex, vector<passivedouble> values) {
+  if (!main_config->GetDeform_Mesh()) {
+    SU2_MPI::Error("Mesh solver is not defined!", CURRENT_FUNCTION);
+  }
+
+  auto iPoint = GetMarkerVertexIndices(iMarker, iVertex);
+
+  if (values.size() != nDim) {
+    SU2_MPI::Error("Invalid number of dimensions!", CURRENT_FUNCTION);
+  }
+
+  for (auto iDim = 0u; iDim < nDim; iDim++) {
+    solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->GetNodes()->SetBound_Disp(iPoint, iDim, values[iDim]);
+  }
+}
+
+vector<vector<passivedouble>> CDriverBase::GetMarkerVertexNormals(unsigned short iMarker, bool normalize) const {
+  const auto nVertex = GetNumberMarkerVertices(iMarker);
+
+  vector<vector<passivedouble>> values;
+
+  for (auto iVertex = 0ul; iVertex < nVertex; iVertex++) {
+    values.push_back(GetMarkerVertexNormals(iMarker, iVertex, normalize = normalize));
+  }
+
+  return values;
+}
+
+vector<passivedouble> CDriverBase::GetMarkerVertexNormals(unsigned short iMarker, unsigned long iVertex,
+                                                          bool normalize) const {
+  if (iVertex >= GetNumberMarkerVertices(iMarker)) {
+    SU2_MPI::Error("Vertex index exceeds marker size.", CURRENT_FUNCTION);
+  }
+
+  vector<passivedouble> values(nDim, 0.0);
+
+  auto normal = main_geometry->vertex[iMarker][iVertex]->GetNormal();
+  auto area = GeometryToolbox::Norm(nDim, normal);
+
+  for (auto iDim = 0u; iDim < nDim; iDim++) {
+    if (normalize) {
+      values[iDim] = SU2_TYPE::GetValue(normal[iDim] / area);
+    } else {
+      values[iDim] = SU2_TYPE::GetValue(normal[iDim]);
+    }
+  }
+
+  return values;
+}
+
+void CDriverBase::CommunicateMeshDisplacements(void) {
+  solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->InitiateComms(main_geometry, main_config, MESH_DISPLACEMENTS);
+  solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->CompleteComms(main_geometry, main_config, MESH_DISPLACEMENTS);
+}
+
+void CDriverBase::ReadFFDInfo(CGeometry* geometry, CConfig* config, CFreeFormDefBox** FFDBox, string val_mesh_filename) {
+  string text_line, iTag;
+  ifstream mesh_file;
+  su2double CPcoord[3], coord[] = {0, 0, 0};
+  unsigned short degree[3], iFFDBox, iCornerPoints, iControlPoints, iMarker, iDegree, jDegree, kDegree, iChar,
+      LevelFFDBox, nParentFFDBox, iParentFFDBox, nChildFFDBox, iChildFFDBox, nMarker, *nCornerPoints, *nControlPoints;
```

Comment:
Copy pasted

Reply:
As discussed in a comment above, I will remove the FFD functions in CDriverBase from the current PR and move it as part of PR #1750. This function will be removed

Reply:
Removed in commit [3c8f5cd](https://github.com/su2code/SU2/pull/1300/commits/3c8f5cd84eea3a870c6f3b0bf2b74fdbdfead498)